services/build_runner_service.rb

```diff
@@ -33,9 +33,9 @@ def build_runner_task_queue
     end
 
     # Fetch all the active runners, and see if there is one WIP
-    def find_build_runner(project_id:, build_number:)
+    def find_build_runner(project_id: nil, sha: nil, build_number: nil)
```

Comment:
I think you might want to leave off `nil` for these params: `project_id: nil, sha: nil, build_number: nil`
that way all of them  are required: `(project_id:, sha:, build_number:)`

Reply:
The idea is that you can either filter by `sha` or `build_number`, being the `project_id` mandatory in all cases (that way I don't break things)

Reply:
Ahh ok, maybe just remove the `nil` for `project_id: nil`?